src/Integration/UnintrusiveBinding/UnintrusiveBindingPathProvider.cs

```diff
@@ -0,0 +1,89 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2023 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.ComponentModel.Composition;
+using System.IO;
+using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudio.Shell.Interop;
+using SonarLint.VisualStudio.Core;
+
+namespace SonarLint.VisualStudio.Integration.UnintrusiveBinding
+{
+    /// <summary>
+    /// Return the path of solution's binding configuration file when in connected mode.
+    /// </summary>
+    internal interface IUnintrusiveBindingPathProvider
```

Comment:
I've deliberately named the new interfaces and classes differently and put them in a separate folder to make the difference between the old and new code more obvious.
The goal in the longer term is to move most of the binding classes to the `ConnectedMode` assembly. For now I've put them in `Integration` along with the old code.